File: weyl-core/src/main/java/io/quartic/weyl/core/catalogue/CatalogueEvent.java
package io.quartic.weyl.core.catalogue;

import io.quartic.catalogue.api.DatasetConfig;
import io.quartic.catalogue.api.DatasetId;
import io.quartic.common.SweetStyle;
import org.immutables.value.Value;

@SweetStyle
@Value.Immutable
public interface CatalogueEvent {
    enum Type {
        CREATE,
        DELETE
    }

    Type type();
    DatasetId id();
    DatasetConfig config();
}

File: catalogue-utils/src/main/kotlin/io/quartic/catalogue/CatalogueEvent.kt
package io.quartic.catalogue

import io.quartic.catalogue.api.model.DatasetConfig
import io.quartic.catalogue.api.model.DatasetCoordinates

data class CatalogueEvent(
        val type: Type,
        val coords: DatasetCoordinates,
        val config: DatasetConfig
) {
    enum class Type {
        CREATE,
        DELETE
    }
}
